lib/taskana-core-test/src/test/java/acceptance/taskcomment/query/QueryTaskCommentAccTest.java

```diff
@@ -0,0 +1,460 @@
+      List<TaskComment> regularResult = taskService.createTaskCommentQuery().list();
+
+      assertThat(offsetAndLimitResult).hasSize(2);
+      assertThat(offsetAndLimitResult.get(0))
+          .isNotEqualTo(regularResult.get(0))
+          .isEqualTo(regularResult.get(1));
+    }
+
+    @WithAccessId(user = ACCESS_USER_1_1_ID)
+    @Test
+    void should_ReturnEmptyList_When_ProvidingWrongConstraints() {
+      List<TaskComment> result = taskService.createTaskCommentQuery().list(0, 1000);
+      assertThat(result).hasSize(3);
+
+      result = taskService.createTaskCommentQuery().list(100, 1000);
+      assertThat(result).isEmpty();
+    }
+  }
+
+  @Nested
+  @TestInstance(Lifecycle.PER_CLASS)
+  class SingleResultTaskComments {
+
+    @WithAccessId(user = ACCESS_USER_1_1_ID)
+    @Test
+    void should_ReturnSingleTaskComment_When_UsingSingleMethod() {
+      TaskComment single =
+          taskService
+              .createTaskCommentQuery()
+              .idIn(comment1.getId())
+              .creatorIn(ACCESS_USER_1_1_ID)
+              .single();
+
+      assertThat(single.getId()).isEqualTo(comment1.getId());
+    }
+
+    @WithAccessId(user = ACCESS_USER_1_1_ID)
+    @Test
+    void should_ThrowException_When_SingleMethodRetrievesMoreThanOneEventFromDatabase() {
+      TaskCommentQuery query = taskService.createTaskCommentQuery().creatorIn(ACCESS_USER_1_1_ID);
+      assertThatThrownBy(query::single).isInstanceOf(TooManyResultsException.class);
+    }
+  }
+
+  @Nested
+  @TestInstance(Lifecycle.PER_CLASS)
+  class TaskCreatorFullNameTaskCommentsWithAdditionalUserInfo
+      implements TaskanaConfigurationModifier {
+
+    @TaskanaInject TaskService taskService;
+    @TaskanaInject UserService userService;
+
+    @Override
+    public TaskanaConfiguration.Builder modify(TaskanaConfiguration.Builder builder) {
+      return builder.addAdditionalUserInfo(true);
+    }
+
+    @WithAccessId(user = ACCESS_USER_1_1_ID)
+    @Test
+    void should_SetTaskCreatorFullNameOfTaskComment_When_PropertyEnabled() throws Exception {
+      List<TaskComment> taskComments =
+          taskService.createTaskCommentQuery().idIn(comment1.getId()).list();
+
+      String creatorFullName = userService.getUser(taskComments.get(0).getCreator()).getFullName();
+      assertThat(taskComments.get(0))
+          .extracting(TaskComment::getCreatorFullName)
+          .isEqualTo(creatorFullName);
+    }
+
+    @WithAccessId(user = ACCESS_USER_1_1_ID)
+    @Test
+    void should_SetTaskCreatorFullNameOfTaskCommentToNull_When_NotExistingAsUserInDatabase() {
+      List<TaskComment> taskComments =
+          taskService.createTaskCommentQuery().idIn(comment3.getId()).list();
+
+      assertThat(taskComments).hasSize(1);
+      assertThat(taskComments.get(0)).extracting(TaskComment::getCreatorFullName).isNull();
+    }
+  }
+
+  @Nested
+  @TestInstance(Lifecycle.PER_CLASS)
+  class TaskCreatorFullNameTaskCommentsWithoutAdditionalUserInfo
```

Comment:
What do you think about CreatorFullNameWhenAdditionalUserInfoIsFalse?

Reply:
I have changed it to TaskCreatorFullNameWhenAdditionalUserInfoIsFalse

What do you think

Reply:
Why TaskCreator? Doesn't the test check for the Creator of the TaskComment?

Reply:
Yes, you are right, I have changed them

Reply:
Changing it to ...CreatorFullNameOfTaskComment... might be misleading because of the FullNameOfTaskComment part. What do you think about "CreatorFullName" without TaskComment? The test is about TaskComments, so it should be clear that the TaskComment is meant

Reply:
I agree, I have changed it to "CreatorFullName"